Index: .idea/modules.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- .idea/modules.xml	(revision )
+++ .idea/modules.xml	(revision )
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectModuleManager">
+    <modules>
+      <module fileurl="file://$PROJECT_DIR$/.idea/aoam.iml" filepath="$PROJECT_DIR$/.idea/aoam.iml" />
+    </modules>
+  </component>
+</project>
\ No newline at end of file
Index: .idea/vcs.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- .idea/vcs.xml	(revision )
+++ .idea/vcs.xml	(revision )
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="VcsDirectoryMappings">
+    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+  </component>
+</project>
\ No newline at end of file
Index: setup/shell/ssh.sh
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- setup/shell/ssh.sh	(revision )
+++ setup/shell/ssh.sh	(revision )
@@ -0,0 +1,10 @@
+#! /bin/sh
+
+username=$1
+if [ -f "/home/$username/.ssh/id_rsa.pub" ];then
+        echo -e "\033[35m SSH已经生成 \033[0m"
+else
+	echo -e "\033[35m 正在生成免密钥SSH \033[0m"
+	ssh-keygen -t rsa -P '' -f /home/$username/.ssh/id_rsa>/dev/null 2>&1
+fi
+
Index: setup/xx.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- setup/xx.txt	(revision )
+++ setup/xx.txt	(revision )
@@ -0,0 +1,2 @@
+xxxx
+172.10.22.57 | SUCCESS | rc=0 >> hadoop [35m SSH已经生成 [0m
Index: setup/ssh.sh
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- setup/ssh.sh	(revision )
+++ setup/ssh.sh	(revision )
@@ -0,0 +1,9 @@
+#! /bin/sh
+
+if [ -f "/home/dianyou/.ssh/id_rsa.pub" ];then
+        echo -e "\033[35m SSH已经生成 \033[0m"
+else
+	echo -e "\033[35m 正在生成免密钥SSH \033[0m"
+	ssh-keygen -t rsa -P '' -f /home/dianyou/.ssh/id_rsa>/dev/null 2>&1
+fi
+
Index: setup/xxx/ssh.sh
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- setup/xxx/ssh.sh	(revision )
+++ setup/xxx/ssh.sh	(revision )
@@ -0,0 +1,10 @@
+#! /bin/sh
+
+username=$1
+if [ -f "/home/$username/.ssh/id_rsa.pub" ];then
+        echo -e "\033[35m SSH已经生成 \033[0m"
+else
+	echo -e "\033[35m 正在生成免密钥SSH \033[0m"
+	ssh-keygen -t rsa -P '' -f /home/$username/.ssh/id_rsa>/dev/null 2>&1
+fi
+
Index: .idea/misc.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- .idea/misc.xml	(revision )
+++ .idea/misc.xml	(revision )
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="ProjectRootManager" version="2" project-jdk-name="Python 3.5.2 (D:\soft\Python35\python.exe)" project-jdk-type="Python SDK" />
+</project>
\ No newline at end of file
Index: setup/proc_memlog.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- setup/proc_memlog.txt	(revision )
+++ setup/proc_memlog.txt	(revision )
@@ -0,0 +1,82 @@
+
+VmRSS:	  136492 kB
+19-11-25 10:11:30
+VmRSS:	  136492 kB
+Cpu: 
+
+19-11-25 10:11:31
+VmRSS:	  136492 kB
+Cpu: 
+
+19-11-25 10:11:32
+VmRSS:	  136492 kB
+Cpu: 
+
+19-11-25 10:11:33
+VmRSS:	  136492 kB
+Cpu: 
+
+19-11-25 10:11:35
+VmRSS:	  136492 kB
+Cpu: 
+
+19-11-25 10:11:36
+VmRSS:	  136492 kB
+Cpu: 
+
+19-11-25 10:11:37
+VmRSS:	  136492 kB
+Cpu: 
+
+19-11-25 10:11:38
+VmRSS:	  136492 kB
+Cpu: 
+
+19-11-25 10:11:39
+VmRSS:	  136492 kB
+Cpu: 
+
+19-11-25 10:12:38
+VmRSS:	  136492 kB
+Cpu:  0.0
+
+19-11-25 10:12:40
+VmRSS:	  136492 kB
+Cpu:  0.0
+
+19-11-25 10:12:41
+VmRSS:	  136492 kB
+Cpu:  0.0
+
+19-11-25 10:12:42
+VmRSS:	  136492 kB
+Cpu:  0.0
+
+19-11-25 10:12:43
+VmRSS:	  136492 kB
+Cpu:  0.0
+
+19-11-25 10:12:44
+VmRSS:	  136492 kB
+Cpu:  0.0
+
+19-11-25 10:12:45
+VmRSS:	  136492 kB
+Cpu:  0.0
+
+19-11-25 10:12:47
+VmRSS:	  136492 kB
+Cpu:  0.0
+
+19-11-25 10:12:48
+VmRSS:	  136492 kB
+Cpu:  0.0
+
+19-11-25 10:12:49
+VmRSS:	  136492 kB
+Cpu:  0.0
+
+19-11-25 10:12:50
+VmRSS:	  136492 kB
+Cpu:  0.0
+
Index: .idea/aoam.iml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- .idea/aoam.iml	(revision )
+++ .idea/aoam.iml	(revision )
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="PYTHON_MODULE" version="4">
+  <component name="NewModuleRootManager">
+    <content url="file://$MODULE_DIR$" />
+    <orderEntry type="inheritedJdk" />
+    <orderEntry type="sourceFolder" forTests="false" />
+  </component>
+  <component name="TestRunnerService">
+    <option name="PROJECT_TEST_RUNNER" value="Unittests" />
+  </component>
+</module>
\ No newline at end of file
Index: setup/source/hosts.ini
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- setup/source/hosts.ini	(revision )
+++ setup/source/hosts.ini	(revision )
@@ -0,0 +1,31 @@
+#
+#C:\Users\MAIBENBEN\AppData\Local\GitHub\PortableGit_284a859b0e6deba86edc624fef1e4db2aa8241a9\git-bash.exe
+[hosts]
+slave1=172.10.22.57
+master=172.10.22.56
+[passwd]
+172.10.22.57=idianyou
+[username]
+user=dianyou
+group=dianyou
+[path]
+dest=/zywa/test/setup/xxx
+src=/zywa/test/setup/shell/
+passwd=idianyou
+master=172.10.22.56
+[softpath]
+#解压安装目录
+jdk_path=/zywa/soft/jdk.tar.gz
+hadoop_path=/zywa/soft/hadoop-2.9.0.tar.gz
+[setuppath]
+jdk_setup_path=/zywa/setup/jdk
+hadoop_setup_path=/zywa/setup/hadoop-2.9.0
+[basepath]
+setuppath=/zywa/setup
+softpath=/zywa/soft
+[master]
+node1=172.10.22.56
+node2=172.10.22.57
+[service]
+node1=NameNode|ResourceManager|SecondaryNameNode
+node2=DataNode|NodeManager
Index: setup/tmp.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- setup/tmp.txt	(revision )
+++ setup/tmp.txt	(revision )
@@ -0,0 +1,6 @@
+Hadoop 2.9.0
+Subversion https://git-wip-us.apache.org/repos/asf/hadoop.git -r 756ebc8394e473ac25feac05fa493f6d612e6c50
+Compiled by arsuresh on 2017-11-13T23:15Z
+Compiled with protoc 2.5.0
+From source with checksum 0a76a9a32a5257331741f8d5932f183
+This command was run using /zywa/setup/hadoop-2.9.0/share/hadoop/common/hadoop-common-2.9.0.jar
\ No newline at end of file
Index: setup/shell/authorized_key.sh
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- setup/shell/authorized_key.sh	(revision )
+++ setup/shell/authorized_key.sh	(revision )
@@ -0,0 +1,12 @@
+#!/usr/bin/expect
+
+set password [lindex $argv 0]
+set username [lindex $argv 1]
+set ip [lindex $argv 2]
+
+spawn scp -r $username@$ip:/home/$username/.ssh/id_rsa.pub /home/$username
+expect "Password:"
+send "$password\r"
+interact
+
+
Index: setup/test.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- setup/test.py	(revision )
+++ setup/test.py	(revision )
@@ -0,0 +1,32 @@
+#!/usr/bin/env python3
+#-*- coding:utf-8 -*-
+#scriptname:py.test.py
+
+import subprocess
+import re
+
+def get_foo():
+    return "foo"
+ 
+def get_bar():
+    return "bar"
+
+def get_data():
+	result = subprocess.run("ansible client -m ping",shell=True)
+	print(result)
+	sub = subprocess.Popen("ansible client -m ping", shell=True, stdout=subprocess.PIPE)
+	content =str(sub.stdout.read(),'utf-8')
+	fo = open("tmp.txt", "w")
+	fo.write(content)
+	fo.close()
+	for line in open("tmp.txt","r"):
+		flag = re.findall("=>",line)
+		if(len(flag)>0):
+			arr = line.split("=>")[0].split("|")
+			if(arr[1].strip() != "SUCCESS"):
+				print(arr[0],"无法通讯")
+
+
+
+if __name__ == "__main__":
+	get_data()
Index: setup/vc.sh
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- setup/vc.sh	(revision )
+++ setup/vc.sh	(revision )
@@ -0,0 +1,13 @@
+#!/bin/bash
+pid=$1  #获取进程pid
+echo $pid
+interval=1  #设置采集间隔
+while true
+do
+    echo $(date +"%y-%m-%d %H:%M:%S") >> proc_memlog.txt
+    cat  /proc/$pid/status|grep -e VmRSS >> proc_memlog.txt    #获取内存占用
+    cpu=`top -n 1 -p $pid|tail -2|head -1|awk '{ssd=NF-4} {print $ssd}'`    #获取cpu占用
+    echo "Cpu: " $cpu >> proc_memlog.txt
+    echo $blank >> proc_memlog.txt
+    sleep $interval
+done
Index: setup/xx.sh
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- setup/xx.sh	(revision )
+++ setup/xx.sh	(revision )
@@ -0,0 +1,25 @@
+#!/bin/bash
+pid=$1 #获取进程pid
+echo $pid
+interval=1 #设置采集间隔
+while true
+do
+echo $(date +"%y-%m-%d %H:%M:%S") >> proc_memlog.txt
+cat /proc/$pid/status|grep -e VmRSS >> proc_memlog.txt #获取内存占用
+cpu=`top -b -n 1 -p $pid  2>&1 | awk -v pid=$pid '{if ($1 == pid)print $9}'` #获取cpu占用
+echo "Cpu: " $cpu >> proc_memlog.txt
+echo $blank >> proc_memlog.txt
+sleep $interval
+done
+
+
+cpu_core=$(grep -c processor /proc/cpuinfo)
+total_time1=$(awk '{if ($1 == "cpu") {sum = $2 + $3 + $4 + $5 + $6 + $7 + $8 + $9 + $10 + $11;print sum}}' /proc/stat)
+cpu_time1=$(awk '{sum=$14 + $15;print sum}' /proc/$pid/stat)
+sleep 1
+total_time2=$(awk '{if ($1 == "cpu") {sum = $2 + $3 + $4 + $5 + $6 + $7 + $8 + $9 + $10 + $11;print sum}}' /proc/stat)
+cpu_time2=$(awk '{sum=$14 + $15;print sum}' /proc/$pid/stat)
+awk -v cpu_time1=$cpu_time1 -v total_time1=$total_time1 -v cpu_time2=$cpu_time2 -v total_time2=$total_time2 -v cpu_core=$cpu_core 'BEGIN{cpu=((cpu_time2 - cpu_time1) / (total_time2 - total_time1)) * 100*cpu_core;print cpu}'
+
+echo $cpu_core $total_time1 $cpu_time1 $total_time2 $cpu_time2 $cpu_core
+
Index: setup/shell/source.sh
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- setup/shell/source.sh	(revision )
+++ setup/shell/source.sh	(revision )
@@ -0,0 +1,6 @@
+#!/bin/sh
+source /home/dianyou/.bashrc
+
+[ $? -eq 0 ] && echo "succ" || echo "Faild"
+
+echo "初始化配置文件..."
Index: setup/hosts.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- setup/hosts.py	(revision )
+++ setup/hosts.py	(revision )
@@ -0,0 +1,285 @@
+#!/usr/bin/python3
+# -*- encoding:utf-8 -*-
+#scriptname:py.hosts.py
+
+import configparser
+import re
+import subprocess
+import os
+import threading
+import time
+
+
+'''
+读取自定义hosts文件
+IP和主机的映射关系
+'''
+config = configparser.ConfigParser()
+config.read("source/hosts.ini")
+
+'''
+修改host文件，IP和域名的映射
+'''
+content = open("/etc/hosts","r").read()
+def check_hosts():
+	num = 0
+	for key in config["hosts"]:
+		hostname = config["hosts"][key] + " " + key
+		if(len(re.findall(hostname,content))>0):
+			#print("\033[32m "+hostname+"已经配置到hosts文件 \033[0m")
+			pass
+		else:
+			num += 1
+			#print("\033[31m "+hostname+"尚未配置到hosts文件 \033[0m")
+			#print("\033[33m 注意：如果检测已经配置，请确认IP和域名是否为一个空格 \033[0m")
+	return num
+
+def get_passwd():
+	dirs="("
+	for key in config["passwd"]:
+		dirs += "["+key+"]=\""+config["passwd"][key]+"\" "
+	dirs += ")"
+	return dirs
+
+def check_ssh():
+	user = config["username"]["user"]
+	rsa = os.path.exists("/home/"+user+"/.ssh/id_rsa.pub")
+	authorized = os.path.exists("/home/"+user+"/.ssh/authorized_keys")
+	if rsa and authorized:
+		return 0
+	else:
+		return 1;
+
+'''
+修改ansible的hosts配置文件
+'''
+def check_ansible_host():
+	content=open("/etc/ansible/hosts").read()
+	_ansibleclient=""
+	if(len(re.findall("client",content))<1):
+        	_ansibleclient="[client]\n"
+
+	for key in config["client"]:
+        	_ip = hostname=config["client"][key]
+        	if(len(re.findall(_ip,content))<1):
+                	_ansibleclient=_ansibleclient+_ip+"\n"
+	return _ansibleclient
+
+
+def check_ansible_client():
+	result = subprocess.run("ansible client -m ping",shell=True)
+	sub = subprocess.Popen("ansible client -m ping", shell=True, stdout=subprocess.PIPE)
+	content =str(sub.stdout.read(),'utf-8')
+	fo = open("tmp.txt", "w")
+	fo.write(content)
+	fo.close()
+
+def make_ansible_ssh():
+	line=get_file_data()
+	array = line[1:len(line)-1].split(" ")
+	ssh = 'ansible client -l '+array[0]+' -m shell -a "'+array[1]+'/ssh.sh '+array[2]+'"'
+	result = subprocess.Popen(ssh, shell=True, stdout=subprocess.PIPE)
+	content =str(result.stdout.read(),'utf-8')
+	fo = open("tmp.txt", "w")
+	fo.write(content)
+	fo.close()
+	ip=""
+	num=0
+	for line in open("tmp.txt","r"):
+		arr=line.split("|")
+		flag=re.findall("生成免密钥SSH",arr[0])
+		if(len(arr)>1):
+			ip=arr[0]
+		elif(len(flag)>0):
+			ssh="shell/authorized_key.sh "+array[5]+" "+array[2]+" "+ip
+			subprocess.run(ssh,shell=True)
+			ssh="cat /home/"+array[2]+"/id_rsa.pub >> /home/"+array[2]+"/.ssh/authorized_keys"
+			subprocess.run(ssh,shell=True)	
+			num +=1
+			print("复制完成...")
+	if(num>0):
+		ssh="ansible "+array[0]+" -m copy -a \"src='/home/"+array[2]+"/.ssh/authorized_keys' dest='/home/"+array[2]+"/.ssh/' owner='"+array[2]+"' group='"+array[3]+"' backup=no\" -o -f 6"
+		subprocess.run(ssh,shell=True)
+	return num
+
+def make_ansible_client():
+	value=0
+	for line in open("tmp.txt","r"):
+		flag = re.findall("=>",line)
+		if(len(flag)>0):
+			arr = line.split("=>")[0].split("|")
+			if(arr[1].strip() != "SUCCESS"):
+				value += 1
+	return value
+
+def get_file_data():
+	value=""
+	for key in config["hosts"]:
+		if(len(value)<1):
+			value="("+config["hosts"][key]
+		else:
+			value += ","+config["hosts"][key]
+
+	value +=" "+config["path"]["dest"]+" "+config["username"]["user"]+" "+config["username"]["group"]+" "+config["path"]["src"]+" "+config["path"]["passwd"]+" "+config["path"]["master"]+")"
+	return value
+
+def setup_jdk():
+	_source=get_file_data()
+	_source=_source[1:len(_source)-1].split(" ")
+	out = subprocess.getoutput("java -version")
+	print("输出内容：",len(re.findall("java version",out)))
+	if(len(re.findall("java version",out))<1):
+		_setup=config["setuppath"]["jdk_setup_path"]
+		_soft=config["softpath"]["jdk_path"]
+		_base=os.path.dirname(_soft)
+		_old_list=os.listdir(_base)
+		_base_path=config["basepath"]["setuppath"]
+		_setup_jdk_path=config["setuppath"]["jdk_setup_path"]
+		print(_setup_jdk_path)
+		if(os.path.exists(_setup)):
+			print("jdk已经解压")
+		else:
+			subprocess.run("tar -zxvf "+_soft+" -C "+_base,shell=True)
+			_new_list=os.listdir(_base)
+			_jdk=_setup_jdk_path.split("/")[len(_setup_jdk_path.split("/"))-1]
+			_jdk_soft_path=_base+"/"+_jdk
+			for ip in _source[0].split(","):
+				_ssh="scp -r "+_jdk_soft_path+" "+_source[2]+"@"+ip+":"+_base_path
+				subprocess.run(_ssh,shell=True)
+			if(len(_jdk_soft_path)>0):
+				subprocess.run("rm -rf "+_jdk_soft_path,shell=True)
+		
+		content = open("/home/"+_source[2]+"/.bashrc","r").read()		
+		if(len(re.findall("JAVA_HOME",content))<1):
+			_line="echo 'export JAVA_HOME="+_setup_jdk_path+"\nexport PATH=$PATH:$JAVA_HOME/bin' >>~/.bashrc"
+			subprocess.run(_line,shell=True)
+		else:
+			print("JDK环境变量已经配置")
+		_ssh="source shell/source.sh "+_source[2]
+		subprocess.run(_ssh,shell=True)
+	else:
+        	print("JDK已经安装：",out)
+
+	
+
+def check(v):
+	return '([key1]="value1" [key2]="value2" [key3]="value3")'
+
+def setup_hadoop():
+	out = subprocess.getoutput("hadoop version")
+	fo = open("tmp.txt", "w")
+	fo.write(out)
+	fo.close()
+	_version=""
+	_num=0
+	for line in open("tmp.txt","r"):
+		_num +=1
+		if(len(_version)<1):
+			_version=line
+	if(_num>1):
+		print("当前Hadoop版本：",_version)
+	else:
+		_source=get_file_data()
+		_source=_source[1:len(_source)-1].split(" ")
+		_soft=config["softpath"]["hadoop_path"]
+		_base=os.path.dirname(_soft)
+		_old_list=os.listdir(_base)
+		_base_path=config["basepath"]["setuppath"]
+		_setup_hadoop_path=config["setuppath"]["hadoop_setup_path"]
+		if(os.path.exists(_setup_hadoop_path)):
+			print("Hadoop已经解压")
+		else:
+			subprocess.run("tar -zxvf "+_soft+" -C "+_base,shell=True)
+			_new_list=os.listdir(_base)
+			_hadoop=_setup_hadoop_path.split("/")[len(_setup_hadoop_path.split("/"))-1]
+			_hadoop_soft_path=_base+"/"+_hadoop
+			_base_path=config["basepath"]["setuppath"]
+			_setup_hadoop_path=config["setuppath"]["hadoop_setup_path"]
+			for ip in _source[0].split(","):
+				_ssh="scp -r "+_hadoop_soft_path+" "+_source[2]+"@"+ip+":"+_base_path
+				subprocess.run(_ssh,shell=True)
+			if(len(_hadoop_soft_path)>0):
+				subprocess.run("rm -rf "+_hadoop_soft_path,shell=True)
+
+		if(len(re.findall("HADOOP_HOME",content))<1):
+			_var=["export HADOOP_HOME=/zywa/setup/hadoop-2.9.0","export PATH=$PATH:$HADOOP_HOME/bin:$HADOOP_HOME/sbin","export LIB_NATIVE_DIR=$HADOOP_HOME/lib/native:$LD_LIBRARY_PATH","export HADOOP_CONF_DIR=$HADOOP_HOME/etc/hadoop"]
+			_line="echo '"+"\n".join(_var)+"' >>~/.bashrc"
+			subprocess.run(_line,shell=True)
+		else:
+			print("Hadoop环境变量已经配置..")
+		_ssh="source shell/source.sh "+_source[2]
+		subprocess.run(_ssh,shell=True)
+
+def source_pro():
+	line=get_file_data()
+	array = line[1:len(line)-1].split(" ")
+	ssh="ansible "+array[0]+" -m copy -a \"src='/home/"+array[2]+"/.bashrc' dest='/home/"+array[2]+"/' owner='"+array[2]+"' group='"+array[3]+"' backup=no\" -o -f 6"
+	subprocess.run(ssh,shell=True)
+	subprocess.run('ansible client -m shell -a "source /home/'+array[2]+'/.bashrc" -f 3',shell=True)
+
+def check_hadoop():
+	master=config["master"]
+	for key in master:
+		ip = config["master"][key]
+		result = subprocess.Popen('ansible client -l '+ip+' -a "jps"', shell=True, stdout=subprocess.PIPE)
+		content =str(result.stdout.read(),'utf-8')
+		#服务列表
+		values = config["service"][key]
+		print(ip+":")
+		for name in values.split("|"):
+			if(len(re.findall(name,content))<1):
+				start_hadoop(ip,name.lower())
+			else:
+				print(name+"服务正在运行")
+		print("\n")
+
+def start_hadoop(ip,name):
+	HADOOP_HOME=config["setuppath"]["hadoop_setup_path"]
+	if(name=="secondarynamenode" or name=="namenode" or name=="datanode"):
+		_shell='ansible client -l '+ip+' -a "'+HADOOP_HOME+'/sbin/hadoop-daemon.sh start '+name+'"'
+		subprocess.run(_shell,shell=True)
+	elif(name=="resourcemanager" or name=="nodemanager"):
+		_shell='ansible client -l '+ip+' -a "'+HADOOP_HOME+'/sbin/yarn-daemon.sh start '+name+'"'
+		subprocess.run(_shell,shell=True)
+
+def job():
+	print("\033[35m 内存使用情况监测 \033[0m")
+	
+	_shell='ansible client -a "free -h"'
+	subprocess.run(_shell,shell=True)
+	print("\033[35m 磁盘使用情况监测 \033[0m")
+	_shell='ansible client -a "df -lh"'
+	subprocess.run(_shell,shell=True)
+	print("\033[35m CPU使用负载监测 \033[0m")
+	_shell="ansible client -a 'uptime'"
+	subprocess.run(_shell,shell=True)
+	print("\033[35m 服务检测 \033[0m")
+	check_hadoop()
+
+def run():
+	while 1:
+		job()
+		time.sleep(10)
+'''
+	scheduler = BlockingScheduler()
+	scheduler.add_job(job,"cron",day_of_week="0-6",minute="*")
+	scheduler.start()
+'''
+if __name__ == "__main__":
+	run()
+	#check_hadoop()
+	#start_hadoop("172.10.22.56","secondarynamenode")
+	#source_pro()
+	#setup_hadoop()
+	#setup_jdk()
+	#make_ansible_ssh()
+	#make_ansible_client()
+	#num = check_ssh()
+	#print(num)
+	#print(check("xxx"))
+	#$get_hosts()
+	#pass
+
+
+
+
Index: setup/start.sh
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- setup/start.sh	(revision )
+++ setup/start.sh	(revision )
@@ -0,0 +1,115 @@
+#! /bin/sh
+
+echo "********************正在检测依赖**************************" 
+count=0
+sshpass -V >> /dev/null
+if [ $? != 0 ];then
+    echo -e "\033[31m 请安装sshpass \033[0m"
+    count=$((${count} + 1))
+else
+    echo -e "\033[32m sshpass已经安装 \033[0m"
+fi
+ansible --version >> /dev/null
+if [ $? != 0 ];then
+    echo -e "\033[31m 请安装ansible \033[0m"
+    count=$((${count} + 1))
+else
+    echo -e "\033[32m ansible已经安装 \033[0m"
+fi
+
+python3 -V >> /dev/null
+if [ $? != 0 ];then
+    echo -e "\033[31m 请安装python \033[0m"
+    count=$((${count} + 1))
+else
+    echo -e "\033[32m python已经安装 \033[0m"
+fi
+
+if [ $count -gt 0 ];then
+	echo -e "\033[31m 缺少 $count 个依赖，请先安装依赖 \033[0m" 
+	exit 1
+else
+	echo -e "\033[32m 依赖检测完毕，准备安装组件..... \033[0m"
+fi
+
+count=`python3 -c 'import hosts; print(hosts.check_hosts())'`
+if [ $count -gt 0 ];then
+	echo -e "\033[31m hosts检测有误,请修复后才能继续 \033[0m"
+	exit 1
+else
+	echo -e "\033[32m hosts检测完毕..... \033[0m"
+fi
+
+declare -A map=`python3 -c 'import hosts; print(hosts.get_passwd())'`
+declare -a user_data=`python3 -c 'import hosts; print(hosts.get_file_data())'`
+
+result=`python3 -c 'import hosts; print(hosts.check_ssh())'`
+
+if [ $result -eq 1 ];then 
+	echo -e "\033[35m 正在生成免密钥SSH \033[0m"
+	ssh-keygen -t rsa -P '' -f ~/.ssh/id_rsa>/dev/null 2>&1
+	cat ~/.ssh/id_rsa.pub >> ~/.ssh/authorized_keys
+	chmod 600 ~/.ssh/authorized_keys
+	for key in $(echo ${!map[*]})
+	do
+		sshpass -p${map[$key]} ssh-copy-id -i ~/.ssh/id_rsa.pub -o StrictHostKeyChecking=no -p22 ${user_data[2]}@$key	
+        	echo "$key : ${map[$key]}"
+	done
+else
+	echo -e "\033[32m SSH密钥已经生成 \033[0m"
+fi
+
+python3 -c 'import hosts; print(hosts.check_ansible_client())'
+
+result=`python3 -c 'import hosts; print(hosts.make_ansible_client())'`
+
+if [ $result -ne 0 ];then
+	echo -e "\033[31m有$result 个IP验证未通过 \033[0m"
+	exit
+else
+	echo -e "\033[32m ansible 配置验证成功... \033[0m"
+fi
+
+echo -e "\033[35m 创建目录存储shell脚本 \033[0m"
+declare -a result=`python3 -c 'import hosts; print(hosts.get_file_data())'`
+ansible ${result[0]} -m file -a 'path='${result[1]}' state=directory owner='${result[2]}' group='${result[3]}' mode=0777 recurse=yes'
+echo -e "\033[35m 正在远程复制shell脚本 \033[0m"
+ansible ${result[0]} -m copy -a "src='${result[4]}' dest='${result[1]}' mode=0777 owner='${result[2]}' group='${result[3]}' backup=no" -o -f 6
+echo -e "\033[35m 执行远程shell脚本生成ssh \033[0m"
+result=`python3 -c 'import hosts; print(hosts.make_ansible_ssh())'`
+array=(${result// / })
+len=${#array[*]}
+num=${array[$len-1]}
+
+if [ $num -ne 0 ];then
+	echo "更新完毕..."
+else
+	echo "无需执行..."
+fi
+
+echo -e "\033[32m 正在安装JDK... \033[0m"
+python3 -c 'import hosts; print(hosts.setup_jdk())'
+
+echo -e "\033[32m 正在安装HADOOP... \033[0m"
+python3 -c 'import hosts; print(hosts.setup_hadoop())'
+
+echo -e "\033[32m 正在检测HADOOP服务... \033[0m"
+python3 -c 'import hosts; print(hosts.check_hadoop())'
+
+#source shell/source.sh
+
+#_result=`ansible client -l ${result[0]} -m shell -a "${result[1]}/ssh.sh ${result[2]}"`
+#echo $_result >> xx.txt
+#通过逗号分隔IP
+#IPS=${result[0]}
+#array=(${IPS//,/ })
+#for ip in ${array[@]}
+#do
+#	echo ${result[5]} ${result[2]} $ip
+#	./shell/authorized_key.sh ${result[5]} ${result[2]} $ip
+#	cat /home/${result[2]}/id_rsa.pub >> /home/${result[2]}/.ssh/authorized_keys
+#done
+
+#echo -e "\033[32m 同步公钥到各个子节点 \033[0m"
+#ansible ${result[0]} -m copy -a "src='/home/${result[2]}/.ssh/authorized_keys' dest='/home/${result[2]}/.ssh/' owner='${result[2]}' group='${result[3]}' backup=yes" -o -f 6
+
Index: setup/xxx/authorized_key.sh
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- setup/xxx/authorized_key.sh	(revision )
+++ setup/xxx/authorized_key.sh	(revision )
@@ -0,0 +1,12 @@
+#!/usr/bin/expect
+
+set password [lindex $argv 0]
+set username [lindex $argv 1]
+set ip [lindex $argv 2]
+
+spawn scp -r $username@$ip:/home/$username/.ssh/id_rsa.pub /home/$username
+expect "Password:"
+send "$password\r"
+interact
+
+
Index: setup/xxx/source.sh
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- setup/xxx/source.sh	(revision )
+++ setup/xxx/source.sh	(revision )
@@ -0,0 +1,8 @@
+#!/bin/sh
+#C:\Users\MAIBENBEN\AppData\Local\GitHub\PortableGit_284a859b0e6deba86edc624fef1e4db2aa8241a9\git-bash.exe
+source /home/dianyou/.bashrc
+
+[ $? -eq 0 ] && echo "succ" || echo "Faild"
+
+echo "初始化配置文件..."
+sleep 10
\ No newline at end of file
